x-pack/solutions/security/plugins/security_solution/public/siem_migrations/rules/service/telemetry.ts

```diff
@@ -0,0 +1,151 @@
+  };
+
+  reportSetupMigrationOpenResources = (params: {
+    migrationId: string;
+    missingResourcesCount: number;
+  }) => {
+    this.telemetryService.reportEvent(SiemMigrationsEventTypes.SetupMigrationOpenResources, params);
+  };
+
+  reportSetupMigrationCreated = (params: {
+    migrationId?: string;
+    rulesCount: number;
+    error?: Error;
+  }) => {
+    const { migrationId, rulesCount, error } = params;
+    this.telemetryService.reportEvent(SiemMigrationsEventTypes.SetupMigrationCreated, {
+      migrationId,
+      rulesCount,
+      ...this.getBaseResultParams(error),
+    });
+  };
+
+  reportSetupResourceUploaded = (params: {
+    migrationId: string;
+    type: RuleMigrationResourceType;
+    count: number;
+    error?: Error;
+  }) => {
+    const { migrationId, type, count, error } = params;
+    this.telemetryService.reportEvent(SiemMigrationsEventTypes.SetupResourcesUploaded, {
+      migrationId,
+      count,
+      type,
+      ...this.getBaseResultParams(error),
+    });
+  };
+
+  reportSetupRulesQueryCopied = (params: { migrationId?: string }) => {
+    const { migrationId } = params;
+    this.telemetryService.reportEvent(SiemMigrationsEventTypes.SetupRulesQueryCopied, {
+      migrationId,
+    });
+  };
+
+  reportSetupMacrosQueryCopied = (params: { migrationId: string }) => {
+    this.telemetryService.reportEvent(SiemMigrationsEventTypes.SetupMacrosQueryCopied, params);
+  };
+
+  reportSetupLookupNameCopied = (params: { migrationId: string }) => {
+    this.telemetryService.reportEvent(SiemMigrationsEventTypes.SetupLookupNameCopied, params);
+  };
+
+  reportStartTranslation = (params: {
+    migrationId: string;
+    connectorId: string;
+    retry?: SiemMigrationRetryFilter;
+    error?: Error;
+  }) => {
+    const { migrationId, connectorId, retry, error } = params;
+    this.telemetryService.reportEvent(SiemMigrationsEventTypes.StartTranslation, {
+      migrationId,
+      connectorId,
+      isRetry: !!retry,
+      ...(retry && { retryFilter: retry }),
+      ...this.getBaseResultParams(error),
+    });
+  };
+
+  // Translated rule actions
+
+  reportTranslatedRuleUpdate = (params: { ruleMigration: RuleMigration; error?: Error }) => {
+    const { ruleMigration, error } = params;
+    this.telemetryService.reportEvent(SiemMigrationsEventTypes.TranslatedRuleUpdate, {
+      migrationId: ruleMigration.migration_id,
+      ruleMigrationId: ruleMigration.id,
+      ...this.getBaseResultParams(error),
+    });
+  };
+
+  reportTranslatedRuleInstall = (params: {
+    ruleMigration: RuleMigration;
+    enabled: boolean;
+    error?: Error;
+  }) => {
+    const { ruleMigration, enabled, error } = params;
+    const elasticRule = ruleMigration.elastic_rule;
+
+    this.telemetryService.reportEvent(SiemMigrationsEventTypes.TranslatedRuleInstall, {
+      migrationId: ruleMigration.migration_id,
+      ruleMigrationId: ruleMigration.id,
+      author: elasticRule?.prebuilt_rule_id ? 'elastic' : 'custom',
```

Comment:
nit, consider raising the logic here out of `reportEvent`

```
const prebuiltRuleConfig = { author: 'elastic' };
if (elasticRule?.prebuilt_rule_id) {
 prebuiltRuleConfig.author = 'custom';
 prebuiltRuleConfig.prebuiltRule = { id: elasticRule.prebuilt_rule_id, title: elasticRule.title };
 }
 
 {
   ...prebuiltRuleConfig
 }
```

Reply:
done here https://github.com/elastic/kibana/pull/209633/commits/671771fbd78d94d03e9ab3e2c93824ab1ed10630